test/e2e/testsuites/pre_provisioned_dangling_attach_volume_tester.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// PreProvisionedDanglingAttachVolumeTest will provision required PV(s), PVC(s) and Pod(s)
+// Testing that a volume could be reattached to a different node on DanglingAttachError
+type PreProvisionedDanglingAttachVolumeTest struct {
+	CSIDriver     driver.PreProvisionedVolumeTestDriver
+	Pod           PodDetails
+	VolumeContext map[string]string
+}
+
+func (t *PreProvisionedDanglingAttachVolumeTest) Run(client clientset.Interface, namespace *v1.Namespace) {
+
+	// Setup required PV(s) and PVC(s) corresponding to PodDetails that will be shared among two pods
+	tpod, cleanup := t.Pod.SetupWithPreProvisionedVolumes(client, namespace, t.CSIDriver, t.VolumeContext)
+	// defer must be called here for PV(s) and PVC(s) to be removed after the test completion and not earlier
+	for i := range cleanup {
+		defer cleanup[i]()
+	}
+
+	// copy tpod for later recreating on a different node
+	tpodcopy := *tpod
+
+	// get a list of available nodes for scheduling the pod
+	nodes := tpod.ListNodes()
+
+	// we need at least 2 nodes to verify the reattach volume
+	if len(nodes) > 2 {
+
+		// assign the pod to the first node, skipping the default scheduling
+		tpod.SetNodeName(nodes[0])
+
+		// create the pod and wait for success
+		ginkgo.By("deploying the pod on node#0")
+		tpod.Create()
+		defer tpod.Cleanup()
+		ginkgo.By("checking that the pod's command exits with no error")
+		tpod.WaitForSuccess()
+
+		// make the tpod's node unschedulable to ensure that the next pod is scheduled on a different node by the default scheduler
+		tpod.MakeNodeUnschedulable(true)        // kubeclt cordon node
```

Comment:
should use Azure api to attach disk, otherwise e2e test won't work since that disk PV is already attached to one node, it won't be allowed to attach to another node(blocked by k8s)